### DISCUSSION TITLE
Weather Warnings to be shown on Main index page.

### DISCUSSION THREAD

**Discussion by JonFear** [Daveiano/weewx-wdc#178] [Ideas]
Hi Dave

Sorry but more work!

With the current amount of weather warnings being issued Europe wide and I guess in the USA and Canada, I wonder whether we should have a clearly visible area showing any weather warnings on the main index page. 

The warnings could perhaps go to the right hand of the "Current Weather conditions" title so it is differentiated from the rest of the tiles and above all, clearly and instantly visible.

Thoughts?

Jon

**Comment by Daveiano** [comment]
Hi Jon,

the skin already includes weather alerts, but only if you use the [weewx-dwd](https://github.com/roe-dl/weewx-DWD) extension. The alerts are shown above the radar tile. I guess this could be a new configuration option:

```ini
# Per default, weather warnings are shown above the radar tile, 
# set to True to show the warnings at the top of the page.
show_warnings_at_top = True
```

Regarding weewx-dwd: it would be best to support other sources for getting the weather warnings (in the best case for worldwide). The weewx-forecast extension could gather this data because some providers (OWM, aeris) are also providing warnings together with the forecast (but the extension currently does not use this information as far as I can tell). 

I will have to do some research on what would be the best option to gather the data!

**Comment by JonFear** [comment]
Hi Dave

Understood. I did not have a good experience with the DWD-Forecast add-in unfortunately. Their English language version is still a work in progress and quite rightly concentrates on the German output.

My previous station software, PWS Dashboard [](https://pwsdashboard.com/) seemed to cope with the world wide Met Warning systems quite well. I wonder if there is anything that can be "borrowed" or gained from their implementation?

BR

Jon

**Comment by Daveiano** [comment]
I could not find any info about what service the PWS Dashboard is using. Do you have any link or source to the "world wide Met Warning system"? A public API is what we need.

**Comment by JonFear** [comment]
HI Dave

Looks like he is using curl on the met office to gather data. I have attached the php files for UK, EU, AU and Curly which is the US sites. My guess is his coverage is not as great as I  thought. I would have thought that CA should be easy to get as it is normally pretty open data there...

A quick google resulted in this which "may" be of some help...
[https://openweathermap.org/api/push-weather-alerts](url)

And this although it almost certainly will cost:
[https://learn.microsoft.com/en-us/rest/api/maps/weather/get-severe-weather-alerts?tabs=HTTP](url)

There other sites but look like they want money...!

Jon
[wrnWarningSamples.zip](https://github.com/Daveiano/weewx-wdc/files/11731490/wrnWarningSamples.zip)

**Comment by Daveiano** [comment]
Ok thank you, I will have a look at the scripts. 

Yes, I also found the OWM API, but it looks like this also has a price tag... (What about https://openweathermap.org/api/one-call-api?)

I will do some more research on what would be best to use.

PS: I also saw your post in GH-140 but did not yet get time for looking into it!

**Comment by OE3MNS** [comment]
Here is an example of my system integration of weather warnings.

[PWS KREUTH](https://www.wetter-kreuth.at/index.html)
I wish you a nice Advent season! 

![kreuth](https://github.com/Daveiano/weewx-wdc/assets/147713099/fe742bcf-5f19-40da-8962-209cb7bc17c1)

**Reply by Daveiano** [reply]
Great work! I think the next release of the skin will include this feature natively, I will see what I can use from the scripts provided here,

Thank you and I wish you also a nice Advent season!
